src/program/graph.rs

```diff
@@ -452,33 +452,36 @@ pub struct ScheduledProgram {
     pub blocks: IndexMap<String, InstructionBlock>,
 }
 
-macro_rules! terminate_working_block {
-    ($terminator:expr, $working_instructions:ident, $blocks:ident, $working_label:ident, $program: ident, $instruction_index: ident) => {{
-        // If this "block" has no instructions and no terminator, it's not worth storing - skip it
-        if $working_instructions.is_empty() && $terminator.is_none() && $working_label.is_none() {
-            $working_label = None
-        } else {
-            let block = InstructionBlock::build(
-                $working_instructions.iter().map(|el| el.clone()).collect(),
-                $terminator,
-                $program,
-            )?;
-            let label =
-                $working_label.unwrap_or_else(|| Self::generate_autoincremented_label(&$blocks));
-
-            match $blocks.insert(label.clone(), block) {
-                Some(_) => Err(ScheduleError {
-                    instruction_index: $instruction_index,
-                    instruction: Instruction::Label(Label(label.clone())),
-                    variant: ScheduleErrorVariant::DuplicateLabel,
-                }), // Duplicate label
-                None => Ok(()),
-            }?;
-            $working_instructions = vec![];
-            $working_label = None
-        }
-        Ok(())
-    }};
+#[allow(unused_assignments)]
+fn terminate_working_block(
+    terminator: Option<BlockTerminator>,
+    working_instructions: &mut Vec<Instruction>,
+    blocks: &mut IndexMap<String, InstructionBlock>,
+    working_label: &mut Option<String>,
+    program: &Program,
+    instruction_index: Option<usize>,
+) -> ScheduleResult<()> {
+    // If this "block" has no instructions and no terminator, it's not worth storing - skip it
+    if working_instructions.is_empty() && terminator.is_none() && working_label.is_none() {
+        *working_label = None;
+    } else {
+        let block = InstructionBlock::build(working_instructions.to_vec(), terminator, program)?;
+        let label = working_label
+            .clone()
+            .unwrap_or_else(|| ScheduledProgram::generate_autoincremented_label(blocks));
+
+        match blocks.insert(label.clone(), block) {
+            Some(_) => Err(ScheduleError {
+                instruction_index,
+                instruction: Instruction::Label(Label(label)),
+                variant: ScheduleErrorVariant::DuplicateLabel,
+            }), // Duplicate label
```

Comment:
With `variant: ScheduleErrorVariant::DuplicateLabel` here, the comment seems a bit superfluous.

Reply:
removed! good catch ✅